modules/statistics-provider-matomo/src/main/java/org/opencastproject/statistics/provider/matomo/StatisticsProviderMatomoService.java

```diff
@@ -0,0 +1,178 @@
+import org.opencastproject.statistics.provider.matomo.provider.MatomoProviderConfiguration;
+import org.opencastproject.statistics.provider.matomo.provider.MatomoTimeSeriesStatisticsProvider;
+import org.opencastproject.util.ConfigurationException;
+
+import org.apache.felix.fileinstall.ArtifactInstaller;
+import org.osgi.service.component.ComponentContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Implements statistics providers using Matomo for data retrieval.
+ */
+@Component(
+    immediate = true,
+    service = { ArtifactInstaller.class },
+    property = {
+        "service.description=Statistics Provider Matomo Service"
+    }
+)
+public class StatisticsProviderMatomoService implements ArtifactInstaller {
+
+  /** Logging utility */
+  private static final Logger logger = LoggerFactory.getLogger(StatisticsProviderMatomoService.class);
+
+  private static final String KEY_MATOMO_API_URL = "matomo.api.url";
+  private static final String KEY_MATOMO_API_TOKEN = "matomo.api.token";
+
+  private String matomoApiUrl;
+  private String matomoApiToken;
+
+  private StatisticsCoordinator statisticsCoordinator;
+  private Map<String, StatisticsProvider> fileNameToProvider = new ConcurrentHashMap<>();
+  private Map<String, BatchMatomoRequest> methodToBatchRequest = new ConcurrentHashMap<>();
+
+  /**
+   * Get an existing batch request for a specific ResourceType and Matomo API method
+   * @param resourceTypeMethod The ResourceType concatenated with the Matomo API method
+   * @return The existing batch request or null if none exists
+   */
+  public BatchMatomoRequest getBatchRequest(String resourceTypeMethod) {
+    return methodToBatchRequest.get(resourceTypeMethod);
+  }
+
+  /**
+   * Register a new batch request for a specific ResourceType and Matomo API method
+   * @param resourceTypeMethod The ResourceType concatenated with the Matomo API method
+   * @param batchRequest The batch request to register
+   */
+  public void registerBatchRequest(String resourceTypeMethod, BatchMatomoRequest batchRequest) {
+    methodToBatchRequest.put(resourceTypeMethod, batchRequest);
+  }
+
+  @Reference
+  public void setStatisticsCoordinator(StatisticsCoordinator service) {
+    this.statisticsCoordinator = service;
+  }
+
+  @Activate
+  public void activate(ComponentContext cc) {
+    logger.info("Activating Statistics Provider Matomo Service");
+  }
+
+  @Deactivate
+  public void deactivate(ComponentContext cc) {
```

Comment:
Other then `ManagedService`, `modified()` isn't called on service activation but only on configuration updates. Calling it yourself will achieve same behavior.

`ComponentContext` removed because it is unused.

```suggestion
  public void activate(Map<String, Object> properties) {
    logger.info("Activating Statistics Provider Matomo Service");
    modified(properties);
  }

  @Deactivate
  public void deactivate() {
```

Reply:
Thanks for pointing that out! I added your suggestions to my last commit.